src/Microsoft.AspNet.OData.Shared/DataModificationExceptionType.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Org.OData.Core.V1
+{   
+    /// <summary>
+    /// Represents a Message Type
+    /// </summary>
+    public class MessageType
+    {
+        /// <summary>
+        /// Code of message
+        /// </summary>
+        public string Code { get; set; }
+
+        /// <summary>
+        /// Actual message
+        /// </summary>
+        public string Message { get; set; }
+
+        /// <summary>
+        /// Severity of message
+        /// </summary>
+        public string Severity { get; set; }
+
+        /// <summary>
+        /// Target of message
+        /// </summary>
+        public string Target { get; set; }
+
+        /// <summary>
+        /// Details of message
+        /// </summary>
+        public string Details { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception Type
+    /// </summary>
+    public abstract class ExceptionType
+    {
+        /// <summary>
+        /// Instanciates a type of <see cref="ExceptionType"/>
+        /// </summary>
+        public ExceptionType()
+        {
+            ODataConventionModelBuilder builder = new ODataConventionModelBuilder();
+            builder.EntityType<DataModificationExceptionType>();
+      
+            builder.Namespace = typeof(DataModificationExceptionType).Namespace;
+
+            Model = builder.GetEdmModel();     
```

Comment:
We should be able to get rid of this.  The Core namespace already defines an ExceptionType (and the derived types) and we should map to those types.